### DISCUSSION TITLE
Choose one group to mesh

### DISCUSSION THREAD

**Discussion by damulaoshi** [neperfepx/neper#738] [Q&A] [answered]
Dear Romain,
First of all, thank you for your open-source software NEPER, it really helps us a lot!

My question is:
Is it possible to mesh only one group in Module-M?
For example, in Module-T, I generate the model as:
neper -T -n from_morpho -morpho "diameq:lognormal(0.1,0.01)+0.5*lognormal(0.01,0.001),1-sphericity:lognormal(0.145,0.03)" -group "diameq>0.03?1:2" -morphooptiini "weight:radeq,coo:packing" -dim 2 -o 2d

We have grains belong to group 1 and group 2. If I want only mesh the grains in group 1, how could I do in Module-M?

Thank you before and wish to hear your response.

Best regards

**Comment by rquey** [comment, marked as answer]
Use `meshface`:
```
 neper -T -n 100 -group "(id<=50)?1:2" -o test -reg 1 -dim 2
 neper -M test.tess -meshface "group==1"
 neper -V test.msh -showelt1d all -dataelt1drad 0.002 -dataelt2dedgerad 0.001 -print img
```
![d](https://github.com/neperfepx/neper/assets/2659376/cb6664d9-0c4f-464b-9935-1f8b6b049e42)

**Reply by damulaoshi** [reply]
Thank you so much for your answer! One more question, is it possible to extend for 3D models? In 3D, I cannot use "-meshface" this command, and how could I change the command?
Thank you before

**Reply by rquey** [reply]
Use `-meshpoly`.

**Reply by damulaoshi** [reply]
Thank you! It is what I want!
